src/game/stage/background/zone_6.c

```diff
@@ -138,8 +138,11 @@ NONMATCH("asm/non_matching/game/stage/background/sub_801D24C.inc",
     stageTime = (gStageTime & 0x3FF);
     stageTime2 = ((gStageTime >> 1) & 0x3FF);
 
-    // p1 *= 2.5
+#ifndef NON_MATCHING
     p1 = (p1 * 2) + HALVE(p1);
+#else
+    p1 *= 2.5;
+#endif
```

Comment:
This is overkill for nonmatching builds.
It would load p1, cast/convert it to float, multiply by 2.5 in software, cast it back to an integer and then store it.
Any platform without an FPU would have to suffer DISPLAY_HEIGHT times per frame with all that.

Reply:
This is a work around because the compiler doesn't like the weird non matching stuff going on in `HALVE` (`p1 = (p1 = p1>> 2`) or something